### PR TITLE
Replace conflicting repository-service-tuf dep

### PR DESCRIPTION
**Pull request by lukpueh**
Previously, `repository-service-tuf` (i.e. the RSTUF cli) was used to bootstrap an RSTUF repo for development. This PR re-implements the relevant parts of the cli locally in Warehouse and removes the `repository-service-tuf` dependency, which conflicts with other dependencies.

Change details
- Add lightweight RSTUF API client library (can be re-used for #15815)
- Add local `warehouse tuf bootstrap` cli subcommand, to wraps lib calls
- Invoke local cli via `make inittuf`
- Remove dependency

supersedes #15958 (cc @facutuesca @woodruffw)

### CODE REVIEW

File: warehouse/tuf/__init__.py
Comment (woodruffw) [inline warehouse/tuf/__init__.py:1]: Code looks fine, but an organization question: are these functions going to be used outside of the `cli.tuf` module? If not IMO we could inline them there.

Comment (lukpueh) [reply warehouse/tuf/__init__.py:1]: Yes, I plan to use the same "RSTUF API library" to "add targets", i.e. tell rstuf to update tuf metadata. See https://github.com/pypi/warehouse/pull/15815

File: warehouse/tuf/__init__.py
Comment (woodruffw) [reply warehouse/tuf/__init__.py:47]: Yeah, this could probably be done with a custom `celery.Task`, although it's been a few years since I've mucked around in Celery 🙂 

IMO this is fine as is, although longer term either a Celery task integration *or* a synchronous wrapper would be ideal.

Comment (lukpueh) [reply warehouse/tuf/__init__.py:47]: I guess I'll leave it as is. My plan for https://github.com/pypi/warehouse/pull/15815 is to already make the calling function (`update_metadata`) an async task, which blocks on `wait_for_success`. So making the latter async as well, maybe isn't helpful. But I'll let the Warehouse maintainers comment.

Thanks for taking a look!

**Review by woodruffw — APPROVED** [review]
LGTM! CC @facutuesca for a review as well, to confirm that this handles the dep issue sufficiently 🙂

**Comment by woodruffw** [comment]
(This will need a rebase/fast-forward onto `main` for a merge.)

**Comment by facutuesca** [comment]
> LGTM! CC @facutuesca for a review as well, to confirm that this handles the dep issue sufficiently 🙂

Yes, this fixes it. Thanks @lukpueh !!

**Review by ewdurbin — APPROVED** [review]

**Review by miketheman — APPROVED** [review]

**Pull request merged**
